### PR TITLE
many small tweaks to the Basics page

### PR DESCRIPTION
**Pull request by SethTisue**
all of these changes are small, don't involve changing ordering,
and are hopefully not controversial.

### CODE REVIEW

**Comment by SethTisue** [comment]
review by @shogowada and community

**Comment by SethTisue** [comment]
there are a few `tut` code blocks that don't compile. @tpolecat I wanted to use `tut:failed` for those, but it didn't work because the code blocks included valid entries as well as invalid entries, so tut complained that _everything_ didn't fail. is that working-as-designed...?

**Comment by tpolecat** [comment]
@SethTisue "correct as implemented" … I think the surprisingly-named `:nofail` will do what you want.

**Comment by SethTisue** [comment]
> I think the surprisingly-named :nofail will do what you want

yeah but it's deprecated. let's discuss further at https://github.com/tpolecat/tut/issues/141

File: tutorials/tour/basics.md [hunk 1]
```diff
@@ -37,51 +37,52 @@ println("Hello," + " world!") // Hello, world!
...
 ### Values
 
-You can name results of expressions by using the `val` keyword.
+You can name results of expressions with the `val` keyword.
 
-```
+```tut
 val x = 1 + 1
-println(x) // 2
+println(x * x) // 4
```
Comment (shogowada) [inline tutorials/tour/basics.md]: Personally, it looks like it's adding unnecessary complication only to show that values hold results of expressions.

What's the reason for this change?

File: tutorials/tour/basics.md [hunk 2]
```diff
@@ -90,33 +91,33 @@ println({
 
 ## Functions
 
-Functions are expressions with parameters.
+Functions are expressions that take parameters and compute a result.
```
Comment (shogowada) [inline tutorials/tour/basics.md]: Does "compute a result" redundant with "Functions are expressions"?

**Comment by shogowada** [comment]
Just for my knowledge, what does `tut` do? I couldn't find anything on the internet or on contributing page.

File: tutorials/tour/basics.md (hunk 1 above)
Comment (SethTisue) [inline tutorials/tour/basics.md]: I wanted the example to have a _reason_ to use `val`, so we are illustrating not only the syntax, but also the use. With my change, introducing `x` now avoids repeating a computation (namely, the adding of 1 and 1).

Maybe I'm overthinking this. I don't have a strong feeling about it.

File: tutorials/tour/basics.md (hunk 2 above)
Comment (SethTisue) [inline tutorials/tour/basics.md]: Ummm... "expressions with parameters" just sounds funny to me. I would never say/write that. Does it mean "functions are values", or does it mean that the body of a function is an expression, or what?

I'm open to other suggestions than the one I've made, but I do think the original sentence is unclear and should be changed _somehow_.

**Comment by SethTisue** [comment]
> what does tut do?

see #591 and #593 

oops, the readme should definitely explain this! I opened https://github.com/scala/scala.github.com/issues/677 on it

**Comment by SethTisue** [comment]
Does anyone have any idea why Travis is failing? I just hit "restart" on the build a second time

File: tutorials/tour/basics.md (hunk 1 above)
Comment (shogowada) [inline tutorials/tour/basics.md]: I see.

How about leaving this line as is and adding another example, maybe something like below?

> You can use values in other expressions.
> ```
> val x = 1 + 1
> println(x * 2) // 4
> ```

File: tutorials/tour/basics.md (hunk 2 above)
Comment (shogowada) [inline tutorials/tour/basics.md]: Oh, I think rephrasing it to "Functions are expressions that take parameters" is an improvement.

By saying "Functions are expressions with parameters", I tried to communicate that functions are, in essence, just expressions with parameters added in its scope. But I think "that take parameters" communicates it clearer.

I just thought, maybe, "and compute a result" is redundant because expressions always compute results.

File: tutorials/tour/basics.md [hunk 3]
```diff
@@ -14,21 +14,21 @@ In this page, we will cover basics of Scala.
 
 ## Trying Scala in the Browser
 
-You can run Scala in your browser by using [ScalaFiddle](https://scalafiddle.io).
+You can run Scala in your browser with [ScalaFiddle](https://scalafiddle.io).
 
 1. Go to https://scalafiddle.io.
```
Comment (shogowada) [inline tutorials/tour/basics.md]: Can you please also link the URL? Automatic link isn't working on the website.
```
1. Go to [https://scalafiddle.io](https://scalafiddle.io).
```

File: tutorials/tour/basics.md (hunk 1 above)
Comment (SethTisue) [inline tutorials/tour/basics.md]: I'll just restore the original.

File: tutorials/tour/basics.md (hunk 2 above)
Comment (SethTisue) [inline tutorials/tour/basics.md]: I'll use your suggested form.

File: tutorials/tour/basics.md (hunk 3 above)
Comment (SethTisue) [inline tutorials/tour/basics.md]: done

File: tutorials/tour/basics.md (hunk 2 above)
Comment (tpolecat) [inline tutorials/tour/basics.md]: Functions *have* parameters and *take* arguments. It probably doesn't matter, but if you're keeping a running list of things that bug me there you go.

**Comment by SethTisue** [comment]
I've responded to all reviewer feedback. not sure what the Travis flakiness is, but so far it always eventually passes if you hit rebuild a few times, so I'm going to go ahead and merge, since this page is getting attention right now since Heather tweeted (and scala_lang retweeted) about it https://twitter.com/heathercmiller/status/829653981089714176

File: tutorials/tour/basics.md (hunk 2 above)
Comment (som-snytt) [inline tutorials/tour/basics.md]: In a function application, are arguments applied to a function or the function to the arguments? Params and args can also be values or types. Functions also reject args when an application fails to typecheck.

Comment (som-snytt) [inline tutorials/tour/basics.md]: For now, I will just think of methods as something like functions.
